src/test/java/com/reto/usuario/infrastructure/entrypoint/UserRestControllerTest.java

```diff
@@ -143,4 +161,93 @@ void test_registerUserAsOwner_withExistingEmailInUserRequestDto_shouldThrowStatu
                 .andExpect(status().isConflict())
                 .andExpect(jsonPath("$.message").value(ExceptionResponse.EMAIL_EXISTS.getMessage()));
     }
+
+    @WithMockUser(username = USERNAME_OWNER, password = PASSWORD, roles = {ROLE_OWNER})
+    @Test
+    void test_registerUserAsEmployee_withAllFieldsCompleteUserRequestToCreateEmployeeDtoAndTokenValid_shouldResponseStatusCreatedAndValueFieldIdUserSavedInTheDataBase() throws Exception {
```

Comment:
Se supone que cada metodo test debe ser independiente pero esta prueba me genera un error cuando ejecuto la clase con todos los test, pero ejecutandolo solo este metodo, esta aprueba: si uso esta anotacion a nivel de clase @DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD) todos los test aprueban pero su problematica es que reinicia el contexto de la aplicacion antes de cada metodo, una solucion puede ser agregar la anotacion @DirtiesContext(methodMode = DirtiesContext.MethodMode.BEFORE_METHOD) encima del test y ya solo se reiniciaria dos veces el contexto de la aplicacion, ejecutando la clase. En el analisis encontre que en la entidad estoy usando en la llave primaria el auto incrementable que me esta provocando que este pedazo de codigo .andExpect(jsonPath("$.idUser").value(2L));  en su respuesta aumente, ya que estoy usando el mismo contexto de la aplicacion.

Reply:
la otra solucion seria como agregarle en el beforeAll que hace antes de cada método un delete * para que vacié la bd

Reply:
Aun sigo teniendo problemas con el metodo aunque ya haya realizado los cambios
    @BeforeAll
    void cleaning() {
        userRepositoryMysql.deleteAll();
   }
el resultado del idUser que espero como Expected: es  2 pero el que obtengo Actual: es 3, no me funciona eliminando los registros porque el valor del idUser esta como auto incrementable y no se elimina ni se reinicia este valor en la base de datos y tampoco se esta relizando un reinicio para este caso, por esta razon me sigue generando que el test falle y ademas por guardar otro usuario en la base de datos en el metodo test_registerUserAsEmployee_withFieldEmailExistsInTheDataBaseAndTokenValid_shouldReturnStatusConflict me genera que otro test que no pertenece a esta PR falle, entonces usando esta anotacion @DirtiesContext(methodMode = DirtiesContext.MethodMode.BEFORE_METHOD) encima del metodo pasa el test 
test_registerUserAsEmployee_withAllFieldsCompleteUserRequestToCreateEmployeeDtoAndTokenValid_shouldResponseStatusCreatedAndValueFieldIdUserSavedInTheDataBase
Pero me falla el test que no pertenece a esta PR, entonces que acciones mas puedo usar para reiniciar el valor del auto incrementable cuando se guarda un usuario.

Reply:
Problema resuelto, se usa JdbcTemplate para realizar el reinicio del indice o auto_increment como tambien se elimina todos los registros usando this.userRepositoryMysql.deleteAll();.
Tambien se agrega la anotacion porque @BeforeAll: se ejecuta una vez al comienzo de la clase. y la anotacion @BeforeEach porque se ejecuta antes de que comience un test.